src/LAMMPS.jl

```diff
@@ -233,6 +260,89 @@ function command(lmp::LMP, cmd::Union{String, Array{String}})
     check(lmp)
 end
 
+"""
+    create_atoms(
+        lmp::LMP, x::Matrix{R1}, id::Vector{I1}, types::Vector{I2};
+        v::Union{Nothing,Matrix{R2}}=nothing,
+        image::Union{Nothing,Vector{I3}}=nothing,
+        bexpand::Bool=false
+    ) where {R1 <: Real, R2 <: Real, I1 <: Integer, I2 <: Integer, I3 <: Integer}
+
+Create atoms for a LAMMPS instance. x contains the atom positions and should be a 3 by n
+Matrix{Float64}, where n is the number of atoms. v contains the associated velocities and 
+should be the same size and type. id, types, and image should all be Vector{Int32} with a 
+length of n.
+"""
+function create_atoms(
+    lmp::LMP, x::Matrix{R1}, id::Vector{I1}, types::Vector{I2};
+    v::Union{Nothing,Matrix{R2}}=nothing,
+    image::Union{Nothing,Vector{I3}}=nothing,
+    bexpand::Bool=false
+) where {R1 <: Real, R2 <: Real, I1 <: Integer, I2 <: Integer, I3 <: Integer}
+    numAtoms = size(x, 2)
+    if size(x, 1) != 3
+        throw(ArgumentError("x must be a n by 3 matrix, where n is the number of atoms"))
+    end
+    if v != nothing && size(x) != size(v)
+        throw(ArgumentError("x and v must be the same size"))
+    end
+    if id != nothing && numAtoms != length(id)
+        throw(ArgumentError("id must have the same length as the number of atoms"))
+    end
+    if types != nothing && numAtoms != length(types)
+        throw(ArgumentError("types must have the same length as the number of atoms"))
+    end
+    if image != nothing && numAtoms != length(image)
+        throw(ArgumentError("image must have the same length as the number of atoms"))
+    end
+
+    jtype = to_julia_type(LAMMPS.extract_atom_datatype(lmp, "x"))
+    if jtype != typeof(x)
+        x = jtype.parameters[1].(x)
+        @warn "Typeof x does not match type expected by LAMMPS. "*
+              "This causes allocation!!! "*
+              "Change typeof x to $(jtype)."
+    end
+
+    jtype = to_julia_type(LAMMPS.extract_atom_datatype(lmp, "id"))
+    if id != nothing && jtype != typeof(id)
+        id = jtype.parameters[1].(id)
+        @warn "Typeof id does not match type expected by LAMMPS. "*
+              "This causes allocation!!! "*
+              "Change typeof id to $(jtype)."
+    end
+
+    jtype = to_julia_type(LAMMPS.extract_atom_datatype(lmp, "type"))
+    if types != nothing && jtype != typeof(types)
+        types = jtype.parameters[1].(id)
+        @warn "Typeof types does not match type expected by LAMMPS. "*
+              "This causes allocation!!! "*
+              "Change typeof types to $(jtype)."
+    end
```

Comment:
Yep, fixed this.

Reply:
Is there a particular reason why you decided to determine the type of the property dynamically instead of just following the types specified in the [documentation](https://docs.lammps.org/Classes_atom.html#_CPPv4N9LAMMPS_NS4Atom7extractEPKc)? So `x` and `v` are always `Matrix{Float64}`, `type` is always `Vector{Int32}`, and `image` is always `Vector{IMAGEINT}` I think this would be a lot cleaner because this would allow us to enforce the correct type using a typeassert in the parameterlist.

To get the correct type of imageint etc. we could also just introduce

```julia
const BIGINT = API.lammps_extract_setting(C_NULL, "bigint") == 4 ? Int32 : Int64
const TAGINT = API.lammps_extract_setting(C_NULL, "tagint") == 4 ? Int32 : Int64
const IMAGEINT = API.lammps_extract_setting(C_NULL, "imageint") == 4 ? Int32 : Int64
```
as constants. (and probably export these as well)

Reply:
I just was not sure that LAMMPS always had the same precision for floats. I thought maybe you could compile different versions with different precision. I will go ahead and change that.

What are BIGINT and TAGINT for, though? Just for completeness?

Reply:
Yes, they are just for completeness. They'll probably come in handy in other places though, so its nice to have them in case we need them later.